pkg/controllers/v1alpha1/dataload/implement.go

```diff
@@ -347,10 +349,31 @@ func (r *DataLoadReconcilerImplement) reconcileFailedDataLoad(ctx cruntime.Recon
 		return utils.RequeueIfError(err)
 	}
 
-	// 2. record event and no requeue
+	// 2. check existence of the targetPath in alluxio  dataLoad completely by pod's logs && record event and no requeue.
 	log.Info("DataLoad failed, won't requeue")
-	jobName := utils.GetDataLoadJobName(utils.GetDataLoadReleaseName(targetDataload.Name))
-	r.Recorder.Eventf(&targetDataload, v1.EventTypeNormal, common.DataLoadJobFailed, "DataLoad job %s failed", jobName)
+	releaseName := utils.GetDataLoadReleaseName(targetDataload.Name)
+	jobName := utils.GetDataLoadJobName(releaseName)
+	job, err := utils.GetDataLoadJob(r.Client, jobName, ctx.Namespace)
+	if err != nil {
+		log.Error(err, "Get DataLoad Job Error", "namespace", ctx.Namespace, "jobName", jobName)
+		return utils.RequeueIfError(err)
+	}
+	podList, err := kubeclient.ListPodsByLabel(ctx, job.Namespace, job.Spec.Template.Labels)
+	if err != nil {
+		log.Error(err, "Get podList Error", "targetDataLoadJob", job.Name)
+		return utils.RequeueIfError(err)
+	}
+	if len(podList.Items) > 0 {
+		pod := podList.Items[0]
+		logstr, err := kubeclient.TailPodLogs(pod.Name, pod.Namespace, 3)
+		if err != nil {
+			log.Error(err, "Tail Pod Logs Error", "targetDataLoadJob", job.Name)
+			return utils.RequeueIfError(err)
+		}
+		if strings.Contains(logstr, "dataLoad failed because some paths not exist") {
+			r.Recorder.Eventf(&targetDataload, v1.EventTypeWarning, common.DataLoadJobFailed, "DataLoad job failed because some paths not exist.")
+		}
+	}
```

Comment:
Since we are trying to issue a event to the targetDataLoad to be more user-friendly, do we have to requeue when something unexpected happen during the process? I'm a little worried about the cases where the `Get` and `List` operations fail repeatedly. This may lay extra pressure to Api-server. Maybe we can ignore such error with a best-effort style?

Reply:
> Since we are trying to issue a event to the targetDataLoad to be more user-friendly, do we have to requeue when something unexpected happen during the process? I'm a little worried about the cases where the `Get` and `List` operations fail repeatedly. This may lay extra pressure to Api-server. Maybe we can ignore such error with a best-effort style?

Yeah,in this case, it may be better to simply ignore such errors.